_source/_posts/2022-02-28-shopping-app-with-node-minio.md

```diff
@@ -0,0 +1,709 @@
+---
+layout: blog_post
+title: Build a Basic Shopping App with Node.js and MinIO
+author: deepa-mahalingam
+by: internal-contributor
+communities: [javascript]
+description: "A tutorial that shows you how to build a basic shopping app with Node.js and MinIO."
+tags: [sign-in-widget, okta, node, minio]
+type: conversion
+image: blog/shopping-app-with-node-minio/cover-image.jpg
+github: https://github.com/oktadev/okta-express-minio-app-example
+tweets:
+- "Learn how to build a basic shopping app with @nodejs and @Minio 🍿"
+---
+
+Node.js has become the staple for e-commerce applications. Typically, e-commerce applications have many images and video assets managed in object storage. As the number of products in an e-commerce store increases, so does the size of the object storage. We'll build our Node.js application with a Kubernetes-friendly object storage service, [MinIO](https://min.io/), to scale with demand and secure this application using the [Okta Sign-In Widget](https://github.com/okta/okta-signin-widget).
+
+This tutorial will guide you through the code to build a simple Node.js Shopping App with the Okta's Sign-In Widget for authentication and a [MinIO Server](https://min.io/) to host the store's assets. We will use Express as our application framework and Handlebars as the view engine. We will use the [MinIO Javascript Client SDK](https://docs.minio.io/docs/javascript-client-quickstart-guide) to fetch the application's image assets from the Minio Server.
+
+{% img
+blog/shopping-app-with-node-minio/home.png
+alt:"The example application"
+width:"800" %}{: .center-image }
+
+
+{% include toc.md %}
+
+**Prerequisites**
+
+This tutorial uses the following technologies but doesn't require any prior experience:
+
+* [Node.js](https://nodejs.org/en/) v16.14.0
+* [MinIO Client (mc)](https://docs.minio.io/docs/minio-client-quickstart-guide)
+* [Minio Server](https://docs.min.io/)
+* [Okta CLI](https://cli.okta.com/) v0.10.0
+
+If you'd like to skip the tutorial and just check out the fully built project, you can go [view it on GitHub](https://github.com/oktadev/okta-express-minio-app-example).
+
+## Scaffold your application
+ 
+Create a new directory for your application and navigate into it.
+
+```sh
+mkdir okta-express-minio-app-example
+cd okta-express-minio-app-example
+```
+
+Use the following command to create a `package.json` file for your application:
+
+```sh
+npm init -y
+```
+
+We need the `express` package for building our application, `express-handlebars` as the template engine and `minio` for programmatically accessing object storage. We need the `@okta/okta-signin-widget` package for authenticating users. We have added `@okta/oidc-middleware` and `express-session` so that you can continue to build this shopping app beyond the basics after this tutorial.
```

Comment:
```suggestion
We need the `express` package for building our application, `express-handlebars` as the template engine, and `minio` for programmatically accessing object storage. We need the `@okta/okta-signin-widget` package for authenticating users. So that you can continue to build this shopping app beyond the basics after this tutorial, we've added `@okta/oidc-middleware` and `express-session`.
```
Readability guideline: avoid starting consecutive sentences with the same word or set of words. This is why I changed the 3rd sentence that started with We ....